File: src/shared/services/token-verify.service.ts
import jwt from 'jsonwebtoken';
import { UnauthorizedException } from '../exceptions/unauthorized.exception';
import { UserRequest } from '../types/user-request';

class TokenVerifyService {
  async execute(token: string, secret?: string): Promise<UserRequest> {
    try {
      const payload: UserRequest = jwt.verify(
        token,
        secret ?? String(process.env.JWT_SECRET),
      ) as UserRequest;
      return payload;
    } catch {
      throw new UnauthorizedException('Token inválido!');
    }
  }
}

export const tokenVerifyService = new TokenVerifyService();

File: src/shared/services/token_verify_service.py
import os
import jwt
from src.shared.exceptions.unauthorized_exception import UnauthorizedException


class TokenVerifyService:
    @classmethod
    def execute(
        self,
        token: str,
        secret: str = os.getenv("JWT_SECRET", ""),
    ):
        try:
            payload = jwt.decode(token, secret, algorithms=["HS256"])
            return payload
        except:
            raise UnauthorizedException("Token inválido!")
